indico/modules/networks/controllers.py

```diff
@@ -0,0 +1,101 @@
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from flask import request, session
+from sqlalchemy.orm import contains_eager
+
+from indico.core.db import db
+from indico.modules.categories.models.categories import Category
+from indico.modules.events.models.events import Event
+from indico.modules.networks import logger
+from indico.modules.networks.forms import IPNetworkGroupForm
+from indico.modules.networks.models.networks import IPNetworkGroup
+from indico.modules.networks.views import WPNetworksAdmin
+from indico.web.util import jsonify_data, jsonify_form, jsonify_template
+from MaKaC.webinterface.rh.admins import RHAdminBase
+
+
+class RHNetworkBase(RHAdminBase):
+    CSRF_ENABLED = True
+
+
+class RHManageNetworks(RHNetworkBase):
+    """Management list for IPNetworks"""
+
+    def _process(self):
+        network_groups = IPNetworkGroup.find().order_by(IPNetworkGroup.name).all()
+        return WPNetworksAdmin.render_template('networks.html', network_groups=network_groups)
+
+
+class RHCreateIPNetworkGroup(RHNetworkBase):
+    """Dialog to create an IPNetworkGroup"""
+
+    def _process(self):
+        form = IPNetworkGroupForm()
+        if form.validate_on_submit():
+            network_group = IPNetworkGroup()
+            form.populate_obj(network_group)
+            db.session.add(network_group)
+            logger.info('Network group %s created by %s', network_group, session.user)
+            return jsonify_data(flash=False)
+        return jsonify_form(form)
+
+
+class RHAdminIPNetworkGroupBase(RHNetworkBase):
+    """Base class for managing in IPNetworkGroup"""
+
+    def _checkParams(self):
+        self.network_group = IPNetworkGroup.find_one(id=request.view_args['network_group_id'])
+
+
+class RHEditIPNetworkGroup(RHAdminIPNetworkGroupBase):
+    """Dialog to edit an IPNetworkGroup"""
+
+    def _process(self):
+        form = IPNetworkGroupForm(obj=self.network_group)
+        if form.validate_on_submit():
+            old_name = self.network_group.name
+            form.populate_obj(self.network_group)
+            if self.network_group.name == old_name:
+                logger.info('Network group %s edited by %s', self.network_group, session.user)
+            else:
+                logger.info('Network group %s edited by %s (old name was "%s")', self.network_group, session.user,
+                            old_name)
+            return jsonify_data(flash=False)
+        return jsonify_form(form)
+
+
+class RHDeleteIPNetworkGroup(RHAdminIPNetworkGroupBase):
+    """Dialog to delete an IPNetworkGroup"""
+
+    def _process_GET(self):
+        query = (self.network_group.in_event_acls
+                 .join(Event)
+                 .order_by(Event.title)
```

Comment:
i usually put the `order_by` call at the very end, but it doesn't really make a difference